DuckDuckGo/Subscription/ViewModel/SubscriptionFlowViewModel.swift

```diff
@@ -315,8 +321,8 @@ final class SubscriptionFlowViewModel: ObservableObject {
         DispatchQueue.main.async {
             self.resetState()
         }
-        if webViewModel.url != URL.subscriptionPurchase.forComparison() {
-            self.webViewModel.navigationCoordinator.navigateTo(url: self.purchaseURL)
+        if webViewModel.url != subscriptionManager.url(for: .purchase).forComparison() {
+            self.webViewModel.navigationCoordinator.navigateTo(url: subscriptionManager.url(for: .purchase))
```

Comment:
We need to load the URL with the origin appended. In the init we’re appending the origin to the purchaseURL variable but not using it anymore.  

```suggestion
             self.webViewModel.navigationCoordinator.navigateTo(url: purchaseURL)
```

Reply:
Merged! Thanks